Update README

cspray opened the pull request:
- Updates the README to be formatted more generally in-line with the
primary amphp/amp package.

- Updates the examples the README to properly return a Promise and use
appropriate method names.

---

I also would like to have a clarifying question around the documentation on `Websocket::onHandshake`, specifically the `@return` documentation:

```
@return Promise<Response> Resolve with the given response to accept the connection or resolve with a new Response object to deny the connection.
```

This phrasing implies that  a brand new Response instance needs to be created to deny the connection. However, the example in the README, and I believe the more intuitive API, is for any non-2XX response to deny the connection. I am unsure as to the exact semantics here but I would like to either clarify the Websocket docblocks or the example to be consistent.

cspray committed:
Update README

- Updates the README to be formatted more generally in-line with the
primary amphp/amp package.

- Updates the examples the README to properly return a Promise and use
appropriate method names.

cspray committed:
minor formatting

bwoebi commented on .gitignore:
```diff
@@ -3,4 +3,5 @@ build
-reports
\ No newline at end of file
+reports
+.idea
```
sigh, why do people always want to add it?
`.idea` shall be part of your own global `.gitignore`; not our responsibility.

cspray committed:
remove .idea from .gitignore

cspray commented:
I understand that the .gitignore was bothersome and I removed it. However, the bigger piece about my question between the discrepancy in the documentation and the example still needs an answer.

trowski commented:
A new instance does not have to be returned. The docs should say that the `Response` instance can be modified to be a non-101 response to effectively deny the connection, though generally a 4xx response would be appropriate. The docs should make it clear that the response object can also have headers added, such as cookies, if desired.

trowski commented on README.md:
```diff
@@ -1,14 +1,35 @@
+
+## Requirements
+
+- PHP 7.0+
```
Should be 7.1.

cspray committed:
New method names, clarification on handshake

Fixes the new readme example to use the new method names that are coming
in RC2.

Adds clarification to the docs for Websocket::handleHandshake to clarify
that changing the status code rejects the connection as opposed to
returning a new Response object.

Fixes the PHP requirement specified in the readme.

cspray committed:
minor formatting

cspray committed:
more clarification on status codes

cspray commented:
@trowski @bwoebi Got the tests updated from master to get those passing and addressed all of the feedback. Believe this is good to be merged in. Please let me know if there's anything else I can add.

kelunik commented on README.md:
```diff
@@ -66,3 +92,7 @@ Loop::run(function () use ($server) {
+
+[amphp/http-server]: https://github.com/amphp/http-server
+[Composer]: https://getcomposer.org
+[Request Handler]: https://amphp.org/http-server/classes/request-handler
```
I prefer these to be inline.

cspray committed:
inline markdown links

kelunik commented:
Thanks!
